sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducerClient.cs

```diff
@@ -330,7 +507,17 @@ protected EventHubProducer()
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public virtual Task CloseAsync(CancellationToken cancellationToken = default) => InnerProducer.CloseAsync(cancellationToken);
+        public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
+        {
+            Closed = true;
+
+            await InnerProducer.CloseAsync(cancellationToken).ConfigureAwait(false);
+
+            if (OwnsConnection)
+            {
+                await Connection.CloseAsync().ConfigureAwait(false);
+            }
```

Comment:
Maybe it would make more sense if we put everything inside a `try` block. In the `finally` block, we could make `Closed = InnerProducer.Closed` to make sure we don't close it in case the task got canceled. If our `AmqpProducer` works as expected, this shouldn't be a problem.

Reply:
That's a good thought.   I want to start with marking as closed, though... .just to preempt any attempts at other operations, but resetting in a `finally` is a great idea.